x/validator/handler.go

```diff
@@ -42,40 +38,31 @@ func NewHandler(keeper Keeper) sdk.Handler {
 		case MsgSignReward:
 			res, err = handleMsgSignReward(ctx, keeper, msg, logEntry)
 		default:
-			errMsg := fmt.Sprintf("Unrecognized validator Msg type: %v", msg.Type())
-			log.Error(errMsg)
-			return sdk.ErrUnknownRequest(errMsg).Result()
+			return nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "unrecognized %s message type: %T", ModuleName, msg)
 		}
-		if err != nil {
-			if res.IsOK() {
-				logEntry.Warn = append(logEntry.Warn, err.Error())
-			} else {
-				logEntry.Error = append(logEntry.Error, err.Error())
-			}
 
+		if err != nil {
+			logEntry.Error = append(logEntry.Error, err.Error())
 		}
-		seal.CommitMsgLog(logEntry)
 
-		if !res.IsOK() {
-			return sdk.ErrInternal(err.Error()).Result()
-		}
-		return res
```

Comment:
removing this could cause unnecessary sync_validator tx failure

Reply:
sdk.Result does not have `IsOK ` anymore.